app/code/Bulbulatory/Recomendations/Controller/Adminhtml/Recommendations/Delete.php

```diff
@@ -0,0 +1,34 @@
+<?php
+namespace Bulbulatory\Recomendations\Controller\Adminhtml\Recommendations;
+
+use \Magento\Framework\App\Action\Context;
+use Magento\Framework\Controller\ResultFactory;
+use Bulbulatory\Recomendations\Model\ResourceModel\Recommendation\CollectionFactory;
+
+class Delete extends \Magento\Backend\App\Action
+{
+    public function __construct(
+        Context $context,
+        collectionFactory $collectionFactory
+    ) {
+        $this->_collectionFactory = $collectionFactory;
```

Comment:
Przypisujesz wartość do `_collectionFactory`, ale nie ma tego nigdzie zadeklarowanego, nie jest też to nigdzie używane więc chyba można skasować. Ogólnie ten konstruktor za dużo nie wnosi, więc chyba cały może być skasowany.